src/schema/schema_alter.c

```diff
@@ -91,6 +111,63 @@ __alter_object(WT_SESSION_IMPL *session, const char *uri, const char *newcfg[])
     return (__alter_apply(session, uri, newcfg, WT_CONFIG_BASE(session, object_meta)));
 }
 
+/*
+ * __alter_tiered --
+ *     Alter a tiered metadata.
+ */
+static int
+__alter_tiered(WT_SESSION_IMPL *session, const char *uri, const char *newcfg[], uint32_t flags)
+{
+    WT_DATA_HANDLE *dhandle;
+    WT_DECL_RET;
+    WT_TIERED *tiered;
+    uint32_t object_id;
+    u_int i;
+    char *value;
+    const char *name;
+
+    dhandle = NULL;
+    value = NULL;
+    name = NULL;
+
+    if (!WT_PREFIX_MATCH(uri, "tiered:"))
+        return (__wt_unexpected_object_type(session, uri, "tiered:"));
+
+    /* Open dhandle exclusively, make sure we actually open it, not just acquire a lock. */
+    LF_CLR(WT_DHANDLE_LOCK_ONLY);
```

Comment:
I don't think this is the right fix for the issue. I think opening the handles could lead to problems.
There is a comment in `schema_alter` that says:
```
and the lock only flag is required because we don't need to
have a handle to update the metadata and opening the handle causes problems when meta
tracking is enabled.
```

Per my comment in the ticket itself, your change here to open the handle is because the `tiers` array is currently parsed and handled in `tiered_open`. The code in that manages the `tiers=` configuration needs to either be refactored or duplicated in the `alter` path so that both `open` and `alter` initialize the `tiers` array like you need.

Reply:
Thank you @sueloverso , good catch! I absolutely missed that statement in the comment. Please see my new fix [c80ec45](https://github.com/wiredtiger/wiredtiger/pull/7924/commits/c80ec45db58f9ea33402d08f5ac55a47cbc8846b)